src/app/clusters/occupancy-sensor-server/occupancy-sensor-server.cpp

```diff
@@ -16,12 +16,158 @@
  */
 
 #include "occupancy-sensor-server.h"
+#include "occupancy-hal.h"
 
-#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/AttributeAccessInterfaceRegistry.h>
+#include <app/EventLogging.h>
+#include <app/data-model/Encode.h>
+#include <app/reporting/reporting.h>
+#include <app/util/attribute-storage.h>
+#include <lib/core/CHIPError.h>
 
-#include "occupancy-hal.h"
+using chip::Protocols::InteractionModel::Status;
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace OccupancySensing {
+
+namespace {
+Structs::HoldTimeLimitsStruct::Type
+    sHoldTimeLimitsStructs[MATTER_DM_OCCUPANCY_SENSING_CLUSTER_SERVER_ENDPOINT_COUNT + CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT];
+
+uint16_t sHoldTime[MATTER_DM_OCCUPANCY_SENSING_CLUSTER_SERVER_ENDPOINT_COUNT + CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT];
+} // namespace
+
+CHIP_ERROR OccupancySensingAttrAccess::Init()
+{
+    VerifyOrReturnError(registerAttributeAccessOverride(this), CHIP_ERROR_INCORRECT_STATE);
+    return CHIP_NO_ERROR;
+}
+
+void OccupancySensingAttrAccess::Shutdown()
+{
+    unregisterAttributeAccessOverride(this);
+}
+
+CHIP_ERROR OccupancySensingAttrAccess::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    VerifyOrDie(aPath.mClusterId == app::Clusters::OccupancySensing::Id);
+
+    switch (aPath.mAttributeId)
+    {
+    case Attributes::FeatureMap::Id:
+        ReturnErrorOnFailure(aEncoder.Encode(mFeature));
+        break;
+    case Attributes::HoldTime::Id: {
```

Comment:
Should add FeatureMap and HoldTime to the list of attributes handled via AttributeAccessInterface in `src/app/zap-templates/zcl/zcl-with-test-extensions.json` and `src/app/zap-templates/zcl/zcl.json`